### ISSUE TITLE
Case statement's implicit object notation with a tuple variable shows syntax error

### ISSUE THREAD

**Issue by Enether** [crystal-lang/crystal#4953]
I was following the tutorial
```crystal
case {1, 1}
when {.even?, .odd?}
  # Matches if value1.even? && value2.odd?
end
```
And I wondered what would happen if I used a tuple variable with a union type. 
It turns out that even without the union type this does not compile at all
```crystal
a = {1, 1}
case a
when {.odd?, .odd?}
  # Matches if value1.even? && value2.odd?
end
```
```shell
Syntax error in test.cr:3: unexpected token: .

when {.odd?, .odd?}
```
Which is strange, since implicit object notation does work on say an `Int32` variable

Turns out this compiles but the statement does not get evaluated. Why is this?
```
a = {1, 1}
case a
when {a[0].odd?, a[1].odd?}
  puts "Hello"
end
```

Crystal 0.23.1 (2017-09-08) LLVM 4.0.1 on MacOS Sierra 10.12.6

**Comment by asterite** [comment]
Yes, this is a current limitation. You can only match against an explicit tuple.

I actually think this feature is complex and not needed, and would like to remove it. Using `if` is usually enough here.

**Comment by Enether** [comment]
It's a fun feature but it'd be nice if we had some sort of error message to indicate what the problem is or at least mention it in the tutorial. I couldn't information about this implicit object notation in the API docs as well

**Comment by lego** [comment]
For the sake of discoverability, the current error message the compiler throws due to this is `"unterminated tuple literal"` (as of 0.24.2).

**Comment by straight-shoota** [comment]
With 0.35.1:  https://carc.in/#/r/a2hy
```cr
a = {1, 1}
case a
when {.odd?, .odd?} # Error: unexpected token: .
end
```
The error message could till be improved for this case.
